### ISSUE TITLE
Not working with compatible URLs called through JavaScript

### ISSUE THREAD

**Issue by gingerbeardman** [diatoming/AlertFilterSafariExtension#5]
I have a bunch of items set up from a while ago that work, `github-mac` is the one I use most often.

Now I just added `slack` but I still see the alert. 

With slack, the link I get from my email is like https://test.slack.com/archives/123FC7UUU/p1234765303119123 and that automatically opens the `slack://` URK without clicking.

Any ideas?

<img width="264" alt="Screen shot 2020-11-19 at 11 36 02" src="https://user-images.githubusercontent.com/49612/99661206-692ed700-2a5b-11eb-971e-b97edf8cc0ce.png">

**Comment by diatoming** [comment]
Please make sure Safari menu "Developer -> Allow Unsigned Extensions" is checked

**Comment by gingerbeardman** [comment]
This is already checked. I also have built the latest source myself, signing both the App and Extension.

![Screen shot 2020-11-19 at 11 47 06](https://user-images.githubusercontent.com/49612/99662393-12c29800-2a5d-11eb-8d20-612d22af0692.png)

**Comment by diatoming** [comment]
I just checked the extension with Slack, and can open it without problem.
The webpage also need to reload after new item added.

**Comment by gingerbeardman** [comment]
I have restarted Safari since. But no change.

The Slack page that opens for you is the same as the one in my screenshot above?

**Comment by diatoming** [comment]
This is the page i tried:
https://api.slack.com/reference/deep-linking

**Comment by gingerbeardman** [comment]
That works for me.

So, the problem is the page I show in my screenshot calls the `slack://` URL scheme through JavaScript, the user does not click on anything (in fact there is no `slack://` URL clickable on this page)

Does AlertFilter take care of this scenario?

**Comment by diatoming** [comment]
If the link is start with slack://, then it should work, otherwise it will not work.

**Comment by gingerbeardman** [comment]
"should work"

Have you seen it working when the link is called through JavaScript?

I am reporting here that it does not work in this scenario.

**Comment by diatoming** [comment]
No, the extension only work with plain link, no JavaScript support

**Comment by gingerbeardman** [comment]
OK thanks.
